### PR TITLE
Added rotate left/right button on toolbar when selecting images

### PR DESCRIPTION
**Pull request by puppetsw**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Related #6861

**Details of Changes**
Add details of changes here.
- Added Rotate left, and Rotate right buttons to toolbar when selecting a image file.
- Added event to ItemManipulationModel to handle refreshing the thumbnail after we rotate the file, and hooked up new ItemManipulationModel event to the GridView layout.
- Added helper to BitmapHelper to rotate the image. 

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
image

### CODE REVIEW

File: src/Files/Views/LayoutModes/GridViewBrowser.xaml.cs [hunk 1]
```diff
@@ -51,6 +51,13 @@ protected override void HookEvents()
             ItemManipulationModel.FocusSelectedItemsInvoked += ItemManipulationModel_FocusSelectedItemsInvoked;
             ItemManipulationModel.StartRenameItemInvoked += ItemManipulationModel_StartRenameItemInvoked;
             ItemManipulationModel.ScrollIntoViewInvoked += ItemManipulationModel_ScrollIntoViewInvoked;
+            ItemManipulationModel.RefreshItemsThumbnailInvoked += ItemManipulationModel_RefreshItemThumbnail;
+
+        }
+
+        private void ItemManipulationModel_RefreshItemThumbnail(object sender, EventArgs args)
+        {
+            ReloadItemIcons();
```
Comment (yaichenbaum) [inline src/Files/Views/LayoutModes/GridViewBrowser.xaml.cs:60]: Is there a way to reload the specific icon?

Comment (puppetsw) [reply src/Files/Views/LayoutModes/GridViewBrowser.xaml.cs:60]: Not that I can see. I could try implement it.

**Comment by puppetsw** [comment]
> image Shouldn't it be these icons??

I couldn't see those in the existing glyphs. I'll double check though.

**Comment by esibruti** [comment]
> > image Shouldn't it be these icons??
> 
> I couldn't see those in the existing glyphs. I'll double check though.

I did not know that. ok ok

File: src/Files/Helpers/BitmapHelper.cs [hunk 2]
```diff
@@ -19,5 +22,31 @@ public static async Task<BitmapImage> ToBitmapAsync(this byte[] data)
...
+        /// <param name="filePath">The file path to the image.</param>
+        /// <param name="rotation">The rotation direction.</param>
+        public static async Task Rotate(string filePath, BitmapRotation rotation)
+        {
+            if (string.IsNullOrEmpty(filePath))
+            {
+                return;
+            }
+
+            var file = await StorageFile.GetFileFromPathAsync(filePath);
```
Comment (d2dyno1) [inline src/Files/Helpers/BitmapHelper.cs:39]: Use `StorageHelpers.ToStorageItem<StorageFile>()` and check if it's not null

Comment (d2dyno1) [reply src/Files/Helpers/BitmapHelper.cs:39]: Also, this won't work with remote storage (phone storage etc.)
You'd have to detect if the function fails, and use fulltrust process

Calling @gave92 for help

Comment (puppetsw) [reply src/Files/Helpers/BitmapHelper.cs:39]: @d2dyno1 When I use the StorageHelpers.ToStorageItem<StorageFile>() method I get a InvalidCastException? from the ToStorageItem method. 

image

image
(Exception here)

System.InvalidCastException: 'Unable to cast object of type 'Files.Filesystem.StorageItems.SystemStorageFile' to type 'Windows.Storage.StorageFile'.'

**Comment by yaichenbaum** [comment]
@puppetsw rotating a photo seems to corrupt them and prevents me from opening it.

File: src/Files/Helpers/BitmapHelper.cs (hunk 2 above)
Comment (gave92) [reply src/Files/Helpers/BitmapHelper.cs]: > Also, this won't work with remote storage (phone storage etc.)

ToStorageItem should work everywhere, including network and phone storage ;)

**Comment by puppetsw** [comment]
> @puppetsw rotating a photo seems to corrupt them and prevents me from opening it.

Hmm strange. What filetype was it?

**Comment by yaichenbaum** [comment]
@puppetsw it was a png.

**Comment by puppetsw** [comment]
> @puppetsw it was a png.

I've adjusted the helper to handle the different encoder types.

**Comment by puppetsw** [comment]
> @puppetsw it was a png.

@yair100 Should be all fixed now.

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**